### PR TITLE
CLC-5209, merged webcast feed includes section and instructor metadata

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/CLC-5209

Work necessary to support Webcast widget work:
* Use MyAcademics::Teaching to fold in section metadata. E.g., instructors authorized to sign up.
* Webcast::Merged is now UserSpecificModel.
* We put nothing in feed when CCN has no webcast. Previously, there was an error message, of sorts. 
* remove list of webcast enabled rooms from front-end api until it's necessary

### CODE REVIEW

File: spec/controllers/mediacasts_controller_spec.rb [hunk 1]
```diff
@@ -25,31 +25,45 @@
...
+              :classes=>[
+                {
+                  :sections=>[
+                    { :ccn=>'85006', :section_number=>'201', :instruction_format=>'LEC' }
+                  ]
+                }
+              ]
+            }
+          ]
+          MyAcademics::Teaching.any_instance.should_receive(:courses_list_from_ccns).once.and_return courses_list
```
Comment (paulkerschen) [inline spec/controllers/mediacasts_controller_spec.rb]: @johncrossman I think as part of the long RSpec syntax overhaul, we're now doing this as `expect_any_instance_of(MyAcademics::Teaching).to receive`...

File: spec/models/canvas/webcast_recordings_spec.rb [hunk 2]
```diff
@@ -18,20 +18,35 @@
...
+              {
+                :sections=>[
+                  { :ccn=>'49982', :section_number=>'101', :instruction_format=>'LEC' },
+                  { :ccn=>'81853', :section_number=>'201', :instruction_format=>'LEC' }
+                ]
+              }
+            ]
+          }
+        ]
+        MyAcademics::Teaching.any_instance.should_receive(:courses_list_from_ccns).once.and_return courses_list
```
Comment (paulkerschen) [inline spec/models/canvas/webcast_recordings_spec.rb]: @johncrossman `expect_any_instance_of` as above

File: spec/models/webcast/merged_spec.rb [hunk 3]
```diff
@@ -1,21 +1,23 @@
...
   context 'a fake proxy' do
     let(:options) { {:fake => true} }
 
     context 'no matching course' do
       let(:feed) do
-        Webcast::Merged.new(2014, 'B', [1], options).get_feed
+        Webcast::Merged.new(ldap_uid, 2014, 'B', [1], options).get_feed
+      end
+      before do
+        MyAcademics::Teaching.should_not_receive :new
```
Comment (paulkerschen) [inline spec/models/webcast/merged_spec.rb]: @johncrossman `expect(MyAcademics::Teaching).not_to receive`

File: spec/models/webcast/merged_spec.rb [hunk 4]
```diff
@@ -26,11 +28,31 @@
...
+                  { :ccn=>'87444', :section_number=>'201', :instruction_format=>'LAB' },
+                  { :ccn=>'87447', :section_number=>'202', :instruction_format=>'LAB' },
+                  { :ccn=>'87432', :section_number=>'001', :instruction_format=>'LEC' },
+                  { :ccn=>'87441', :section_number=>'002', :instruction_format=>'LEC' }
+                ]
+              }
+            ]
+          }
+        ]
+        MyAcademics::Teaching.any_instance.should_receive(:courses_list_from_ccns).once.and_return courses_list
```
Comment (paulkerschen) [inline spec/models/webcast/merged_spec.rb]: @johncrossman And one more...

**Comment by johncrossman** [comment]
Thanks @paulkerschen  Fixes made. Applied same change to course_media_spec.rb

**Pull request merged**